drivers/mbox/mbox_nrf_bellboard_local.c

```diff
@@ -0,0 +1,217 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT nordic_nrf_bellboard_local
+
+#include <zephyr/devicetree.h>
+#include <zephyr/drivers/mbox.h>
+
+#include <haly/nrfy_bellboard.h>
+
+#define BELLBOARD_IRQ_COUNT 4U
+
+/* BELLBOARD runtime resources for each group */
+struct mbox_bellboard_local_group {
+	mbox_callback_t cb[NRF_BELLBOARD_EVENTS_TRIGGERED_COUNT];
+	void *user_data[NRF_BELLBOARD_EVENTS_TRIGGERED_COUNT];
```

Comment:
There is no need to use separate arrays of callbacks and user data pointers for particular groups. Just one instance of each is enough.

Reply:
fixed, good catch